dmscripts/email_engine/__init__.py

```diff
@@ -208,10 +208,9 @@ def send_email_notification(
         def send_email_notification(
             notification: EmailNotification,
         ) -> NotificationResponse:
-            # note that all notifications will have the same reference
             return notify_client.send_email_notification(
                 **notification,
-                reference=reference,
+                reference=f"{reference}-{notification.sha256_hash}",
```

Comment:
Do we still want the reference prefix in this case? Would it make future discovery harder if you lost the reference but could derive the hash?

Reply:
I think we do want the reference, because then in our logs we can see what script sent the email as well as the hash of the data sent to Notify.

Reply:
Gideon - I believe both Kibana and Splunk are clever enough to find "prefix-reference" if you search for "reference"